File: backend/app/schemas/system.py
from typing import Optional, List, Any
from pydantic import BaseModel
from app.schemas.common import ItemBase


class SystemLogItem(ItemBase):
    id: str
    user_id: Optional[str] = None
    operation: str
    resource_type: str
    resource_id: Optional[str] = None
    description: Optional[str] = None
    created_at: str


class NotificationItem(ItemBase):
    id: str
    title: str
    content: str
    is_read: bool
    created_at: str


class ConfigUpdate(BaseModel):
    key: str
    value: Any
    description: Optional[str] = None

File: frontend/src/api/system.ts
import request from '@/utils/request'

export interface SystemLogItem {
  id: string
  user_id: string | null
  operation: string
  resource_type: string
  resource_id: string | null
  description: string | null
  created_at: string
}

export interface NotificationItem {
  id: string
  title: string
  content: string
  is_read: boolean
  created_at: string
}

export function getSystemLogs(params: any) {
  return request.get<any, any>('/system/logs', { params })
}

export function getNotifications(params: any) {
  return request.get<any, any>('/system/notifications', { params })
}

export function markNotificationRead(id: string) {
  return request.put(`/system/notifications/${id}/read`)
}

export function getSystemConfig() {
  return request.get('/system/config')
}

export function updateSystemConfig(data: any) {
  return request.put('/system/config', data)
}

export function getSystemStats() {
  return request.get('/system/stats')
}
